persist-tool/dump.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019 Balabit
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "dump.h"
+
+static void
+print_struct_json_style(gpointer data, gpointer user_data)
+{
+  PersistEntryHandle handle;
+  gsize size;
+  guint8 result_version;
+
+  PersistTool *self = (PersistTool *)user_data;
+  gchar *name = (gchar *)data;
+
+  if (!(handle = persist_state_lookup_entry(self->state, name, &size, &result_version)))
+    {
+      fprintf(stderr,"Can't lookup for entry \"%s\"\n", name);
+      return;
+    }
+
+  gpointer block = persist_state_map_entry(self->state, handle);
+
+  fprintf(stderr,"\n%s = { \"value\": \"", name);
```

Comment:
The actual expected payload should go to the `stdout`.

Reply:
done